packages/block-editor/src/components/font-appearance-control/index.js

```diff
@@ -3,72 +3,126 @@
  */
 import { CustomSelectControl } from '@wordpress/components';
 import { useMemo } from '@wordpress/element';
-import { __ } from '@wordpress/i18n';
+import { __, sprintf } from '@wordpress/i18n';
 
 /**
  * Control to display unified font style and weight options.
  *
- * @param  {Object}   props          Component props.
- * @param  {Object}   props.value    Currently selected combination of font style and weight.
- * @param  {Object}   props.options  Object containing weight and style options.
- * @param  {Function} props.onChange Handles selection change.
- * @return {WPElement}               Font appearance control.
+ * @param  {Object}   props Component props.
+ * @return {WPElement}      Font appearance control.
  */
-export default function FontAppearanceControl( { value, options, onChange } ) {
-	const { fontStyle, fontWeight } = value;
-	const { fontStyles = [], fontWeights = [] } = options;
+export default function FontAppearanceControl( props ) {
+	const {
+		disableFontStyles,
```

Comment:
I've changed the control to check the presence of the appropriate options to determine if styles or weights are disabled. 

Given the component that uses this control already checks if the relevant supports have been disabled it can adjust the options passed through to the control as required.